spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -124,11 +125,33 @@ public List<ServiceInstance> getInstances(String serviceId) {
 	}
 
 	public List<Endpoints> getEndPointsList(String serviceId) {
-		return this.properties.isAllNamespaces()
+		List<Endpoints> endpoints = this.properties.isAllNamespaces()
 				? this.client.endpoints().inAnyNamespace().withField("metadata.name", serviceId)
 						.withLabels(properties.getServiceLabels()).list().getItems()
 				: this.client.endpoints().withField("metadata.name", serviceId)
 						.withLabels(properties.getServiceLabels()).list().getItems();
+
+		if (properties.getFilter() == null || properties.getFilter().isEmpty()) {
+			return endpoints;
+		}
+
+		List<Endpoints> result = new ArrayList<>();
```

Comment:
the logic here is verbatim from the PR that is supposed to go in `main`.

P.S. I honestly have no idea how you do your magic in merging `2.1.x` into `main`, it does not look trivial to me at all

Reply:
Oh its not trivial at all!

Reply:
And I dont know why you have to do that. I actually never understood the real need for this. We can always create two PRs for separate branches...